src/library/scala/collection/View.scala

```diff
@@ -364,7 +364,12 @@ object View extends IterableFactory[View] {
         i += 1
         value
       }
-      def hasNext: Boolean = it.hasNext
+      def hasNext: Boolean =
+        if(it.hasNext) true
```

Comment:
There are some places in the compiler that use boolean operator's short-circuit as short-hand for control-flow. Using those, it would write this as: 
```Scala
it.hasNext ||  index >= i && (throw new IndexOutOfBoundsException(index.toString))
```
But, personally, I am not too keen with mixing value functions and control-flow statements.

Reply:
Code like this makes me look for the `:scream:` emoji. Does anyone find that more readable? While `if(x) true else false` is not aesthetically pleasing I find it much easier to understand than side effects hiding in boolean sub-expressions and depending on the correct evaluation order.